python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -822,6 +830,100 @@ async def _wait_for_mtm1m3_detailed_state(
                 )
             self.log.debug(f"M1M3 detailed state {m1m3_detailed_state.detailedState!r}")
 
+    async def _wait_hard_point_test_ok(self, hp: int) -> None:
+        """Wait until the hard point test for the specified hard point
+        finishes.
+
+        Parameters
+        ----------
+        hp : `int`
+            Index of the hard point (starting from 1).
+
+        Raises
+        ------
+        RuntimeError
+            If the hp test failed.
+        """
+
+        self.log.info("Checking if the hard point breakaway test has passed.")
+
+        while True:
+            hp_test_state = MTM1M3.HardpointTest(
+                (
+                    await self.rem.mtm1m3.evt_hardpointTestStatus.next(
+                        flush=False, timeout=self.long_timeout
+                    )
+                ).testState[hp - 1]
+            )
+
+            if hp_test_state == MTM1M3.HardpointTest.FAILED:
```

Comment:
You will not see that in tests, but will get hit when running with HW. The problem is it takes some time for hardpoints to transition from "PASSED" to "being tested state".. And "PASSED"/"FAILED" aren't clear when you issue testHardpoint. Please check current (need to pull - this is a bit new) develop branch of ts_m1m3supporttesting - the wait for bump test in M13T002 is as follow:

```
                if last_test_state is not None and last_test_state != test_state:
                    last_test_state = None
                if test_state in (MTM1M3.BumpTest.PASSED, MTM1M3.BumpTest.FAILED):
                    if last_test_state is None:
                        break
```

when last_test_state is retrieved before the test command is run, only that fully guarantees the code will wait for completion (e.g. see translation from pervious PASSED/FAILED to being tested and then to PASSED/FAILED). That shall be doe to both hardpoint and bump tests - perhaps pass last state(s) as arguments to wait function.

Reply:
Not sure I understand this... I think the problem is that in your M13T002 script you are not flushing the event before sending the command so you might be getting old samples that were issued before the command was issue. You also use a pretty short timeout (1s) whereas I am using a long timeout in waiting for the `next` event. 

Basically what I do is:

1 - flush the event queue
2 - issue the command
3 - wait for the next event (this will only return values that were published after the command was sent).